app/models/commit_status.rb

```diff
@@ -5,8 +5,13 @@ class CommitStatus
   # See https://developer.github.com/v3/repos/statuses for api details
   # - fatal is our own state that blocks deploys
   # - missing is our own state that means we could not determine the status
-  STATE_PRIORITY = [:success, :pending, :missing, :failure, :error, :fatal].freeze
+  STATE_PRIORITY = [:pending, :success, :missing, :failure, :error, :fatal].freeze
```

Comment:
that looks wrong ... if something is pending then it cannot be marked as success